core/chains/evm/txmgr/orm.go

```diff
@@ -263,3 +280,21 @@ func loadEthTxesAttemptsReceipts(q pg.Queryer, etxs []*EthTx) (err error) {
 	}
 	return nil
 }
+
+func loadConfirmedAttemptsReceipts(q pg.Queryer, attempts []EthTxAttempt) error {
+	byHash := make(map[common.Hash]*EthTxAttempt, len(attempts))
+	hashes := make([][]byte, len(attempts))
+	for i, attempt := range attempts {
+		byHash[attempt.Hash] = &attempts[i]
+		hashes = append(hashes, attempt.Hash.Bytes())
+	}
+	var receipts []EthReceipt
+	if err := q.Select(&receipts, `SELECT * FROM eth_receipts WHERE tx_hash = ANY($1)`, pq.Array(hashes)); err != nil {
+		return errors.Wrap(err, "loadConfirmedAttemptsReceipts failed to load eth_receipts")
+	}
+	for _, receipt := range receipts {
+		attempt := byHash[receipt.TxHash]
+		attempt.EthReceipts = append(attempt.EthReceipts, receipt)
+	}
+	return nil
+}
```

Comment:
I think you can get rid of this if you only select one, and maybe even combine it all into one query.